zilliqa/src/p2p_node.rs

```diff
@@ -255,12 +251,37 @@ impl P2pNode {
                         SwarmEvent::NewListenAddr { address, .. } => {
                             info!(%address, "P2P swarm listening on");
                         }
+                        // this is necessary - https://docs.rs/libp2p-kad/latest/libp2p_kad/#important-discrepancies
+                        SwarmEvent::Behaviour(BehaviourEvent::Identify(identify::Event::Received { peer_id, info, .. })) => {
+                            debug!(%peer_id, ?info, "identify event");
+                            if info.protocols.iter().any(|p| *p == kad::PROTOCOL_NAME) {
```

Comment:
Should we remove the protocol check now we are adding peers to the whole swarm?

Reply:
And maybe add it in the `NewExternalAddrOfPeer` handler instead?

Reply:
Or I guess we could:
1. Call `kademlia.add_address` **here** if the protocol matches
2. Call `swarm.add_peer_address` unconditionally here.
?